src/sugar3/graphics/popwindow.py

```diff
@@ -0,0 +1,204 @@
+# Copyright (C) 2016 Abhijit Patel
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+
+from gi.repository import Gtk
+from gi.repository import Gdk
+from gi.repository import GdkX11
+
+from sugar3.graphics import style
+from sugar3.graphics.toolbutton import ToolButton
+from sugar3.graphics.toolbarbox import ToolbarBox
+
+from jarabe.model import shell
+
+
+class PopWindow(Gtk.Window):
+    """
+    UI interface for activity Pop-up Windows.
+    PopWindows are the windows that open on the top of the current window.
+    These pop-up windows don't cover the whole screen.
+    They contain canvas content, alerts messages, a tray and a
+    toolbar.
+
+    Kwargs:
+        title (gettext.gettext or _() ): title label of the pop-up window
+        set_titlebox (bool): False if titlebox is not needed
+        window_xid (xlib.Window): xid of the parent window
+    """
+
+    def __init__(self, title=None, set_titlebox=True, window_xid=None):
+        Gtk.Window.__init__(self)
+        self._title = title
+        self._set_title_box = set_titlebox
+        self._parent_window_xid = window_xid
+
+        self.set_decorated(False)
+        self.set_position(Gtk.WindowPosition.CENTER_ALWAYS)
+        self.set_border_width(style.LINE_WIDTH)
+        self.set_has_resize_grip(False)
+
+        self.connect('realize', self.__realize_cb)
+        self.connect('key-press-event', self.__key_press_event_cb)
+        self.connect('hide', self.__hide_cb)
+
+        self._vbox = Gtk.VBox()
+        self.add(self._vbox)
+        self._vbox.show()
+
+        if self._set_title_box:
+            self._title_box = TitleBox(title)
+            self._title_box.close_button.connect(
+                'clicked',
+                self.__close_button_clicked_cb)
+            self._title_box.set_size_request(-1, style.GRID_CELL_SIZE)
+            self._vbox.pack_start(self._title_box, False, True, 0)
+            self._title_box.show()
+
+        width = Gdk.Screen.width() - style.GRID_CELL_SIZE * 5
+        height = Gdk.Screen.height() - style.GRID_CELL_SIZE * 3
```

Comment:
We really should provide an api to add sizes.  You could use add a GObject property to control them.

Maybe also include some constants for the sizes, like:

`PopWindowSize.FULLSCREEN` = like the view source window
`PopWindowSize.HALF_WIDTH` = like the bibliography "add item" window